lib/network_utils.ml

```diff
@@ -919,7 +919,13 @@ end = struct
         []
       )
     in
-    let request = minimal @ set_gateway @ set_dns in
+    let ipv6_fields =
+      if ipv6 then
+        ["dhcp6.client-id"; "dhcp6.domain-search"; "dhcp6.name-servers"]
+      else
+        []
+    in
+    let request = minimal @ set_gateway @ set_dns @ ipv6_fields in
```

Comment:
It's better to avoid chained appends (`@`) like this and use List.flatten instead. This is because of the depth in recursion is much higher in the chained appends.

```suggestion
    let request = List.flatten [minimal; set_gateway; set_dns; ipv6_fields] in
```